lib/awful/tag.lua

```diff
@@ -741,14 +821,47 @@ function tag.object.set_layout(t, layout)
             t.dynamic_layout_cache[layout] = instance
         end
 
+        template = layout
         layout = instance
     end
 
     tag.setproperty(t, "layout", layout)
 
+    local override = tag.getproperty(t, "layouts")
+
+    if not override then
+        t:emit_signal("property::layouts")
```

Comment:
Uhm, why is the signal emitted here already? Why is it emitted at all? I fail to see how the list of layouts changes here (or at least, always changes here). Shouldn't the following `update_layouts` emit this signal if needed?